binary_search_tree/binary_search_tree.py

```diff
@@ -12,38 +10,87 @@ def __init__(self, value):
 
     # Insert the given value into the tree
     def insert(self, value):
-        pass
+        if value < self.value:
+            if self.left:
+                self.left.insert(value)
+            else:
+                self.left = BinarySearchTree(value)
+        else:
+            if self.right:
+                self.right.insert(value)
+            else:
+                self.right = BinarySearchTree(value)
 
     # Return True if the tree contains the value
     # False if it does not
     def contains(self, target):
-        pass
+        if self.value == target:
+            return True
+        elif self.value > target:
+            if self.left is not None:
+                return self.left.contains(target)
+            else:
+                return False
+        else:
+            if self.right is not None:
+                return self.right.contains(target)
+            else:
+                return False
```

Comment:
This is one of those weird cases where the iterative solution is cleaner than the recursive solution. How might you implement this iteratively?